tests/test_smtp.py

```diff
@@ -0,0 +1,156 @@
+import os
+
+from httpreplay.reader import PcapReader
+from httpreplay.smegma import TCPPacketStreamer, TLSStream
+from httpreplay.cut import smtp_handler
+from httpreplay.cobweb import SmtpProtocol
+
+
+class SmtpTest(object):
+    handlers = {
+        587: smtp_handler,
+        25: smtp_handler
+    }
+
+    def __init__(self, pcapfile):
+        self.f = open(pcapfile, "rb")
+        self.reader = PcapReader(self.f)
+        self.reader.raise_exceptions = False
+        self.reader.tcp = TCPPacketStreamer(self.reader, self.handlers)
+
+
+    def get_sent_recv(self):
+
+        s = None
+        r = None
+        for s, ts, prot, sent, recv in self.reader.process():
+            s = sent
+            r = recv
+
+        self.f.close()
+        return s, r
+
+
+def test_read_smtp_from_to():
+    test = SmtpTest(os.path.join("tests", "pcaps", "smtp-auth-login.pcap"))
+    sent, recv = test.get_sent_recv()
+
+    expected = [
+        ['xxxxxx@xxxxx.co.uk'],
+        ['xxxxxx.xxxx@xxxxx.com']
+    ]
+
+    output = [
+        sent.mail_from,
+        sent.mail_to
+    ]
+    assert expected == output
+
+
+def test_read_smtp_headers():
+    test = SmtpTest(os.path.join("tests", "pcaps", "smtp-auth-login.pcap"))
+    sent, recv = test.get_sent_recv()
+
+    expected = [
+        'Reply-To: <xxxxxx@xxxxx.co.uk>',
```

Comment:
Regarding headers, I was wondering if we should turn those into a dictionary. Or at least a tuple (i.e., in this case, `("Reply-To", "<xxx....uk>")`. Not sure if that's desired here, though. Please judge on your own :)